### PR TITLE
Fix retry count not displaying on pause overlay

### PR DESCRIPTION
**Pull request by DevSDK**
I've find "You've retried xx time(s)" message that something weird.
That is not displayed pause overlay and only see count on FailOverlay
I change that code that PauseContainer.Retries property can be set call-back function.

### CODE REVIEW

**Comment by peppy** [comment]
A more correct fix for this would be to resolve the implementation in `GaameplayMenuOverlay`:

image

**Comment by DevSDK** [comment]
Why? I want more detail information..
I don't know what is wrong with my PR.
Could you advice to be about detail of your suggestion.

**Comment by peppy** [comment]
The issue is that setting `Retries` too early sets nothing, due to that null check in my screenshot. This is incorrect – properties should set their values regardless of the instance's state.

A possible solution would be storing retries to a local `private int retries` in the case the container is null, then perform initialisations in a separate method that is called both in that property (in the non-null case) and also in `load()` or `LoadComplete()`

**Comment by DevSDK** [comment]
Well I've try to change my code.
If ```retryCounterContainer```is null value or not, always storing ```private int retries``` .
So, Retries property set value by ```retries``` member value  and that point ```retryCounterContainer``` is not set null.
I hope my code change is right.
Thx your opinion.
:)

File: osu.Game/Screens/Play/GameplayMenuOverlay.cs [hunk 1]
```diff
@@ -122,13 +122,15 @@ private void load(OsuColour colours)
...
-            Retries = 0;
+            Retries = retries;
         }
 
+        private int retries;
         public int Retries
         {
             set
             {
+                retries = value;
```
Comment (peppy) [inline osu.Game/Screens/Play/GameplayMenuOverlay.cs]: Generally, properties should start with an equality check, in this case
`if (value == retries) return`

I think you should do this here and then create follow it with:

```
retries = value;
if (retryCounterContainer != null)
    updateRetryCount();
```

then change the code in `load()` to call `updateRetryCount()`

**Review by peppy — DISMISSED** [review]
as proposed

**Comment by DevSDK** [comment]
Oh well,

**Review by peppy — APPROVED** [review]

**Pull request merged**